src/main.rs

```diff
@@ -29,55 +27,43 @@ fn main() -> Result<(), Box<dyn Error>> {
 
     let static_frame  = conf.fixed_frame;
 
-    let current_tf = Arc::new(RwLock::new(
-            rosrust_msg::geometry_msgs::Transform::default()));
-    let cb_tf = current_tf.clone();
 
-    println!("spawning tf listener");
-    let listener = Arc::new(Mutex::new(rustros_tf::TfListener::new()));
-    let tf_listener = listener.clone();
-    let sleep = 1000 / conf.target_framerate as u64;
-    let _static_frame = static_frame.clone();
-    let _tf_handle = thread::spawn(move|| {
-        // update robot position thread
-        while rosrust::is_ok() {
-            {
-            let tf_listener = tf_listener.lock().unwrap();
-            let res = &tf_listener.lookup_transform(&_static_frame,
-                                                    "base_link",
-                                                    rosrust::Time::new());
-            if res.is_ok() {
-                let mut cb_tf = cb_tf.write().unwrap();
-                *cb_tf = res.as_ref().unwrap().transform.clone();
-                }
-            }
-            thread::sleep(Duration::from_millis(sleep));
-        }
-    });
+    // Initialize listener and wait for it to come up
+    let listener = Arc::new(rustros_tf::TfListener::new());
+    while rosrust::is_ok() {
```

Comment:
this helps 👍

Reply:
wait forever is ok?

Reply:
This just waits for the buffer to fill,
I can print something but think its fine.